tests/fixtures/watcher_fixtures.py

```diff
@@ -0,0 +1,168 @@
+import pytest
+from pathlib import Path
+import shutil
+import time
+from datetime import datetime
+from iSparrow.utils import wait_for_file_completion
+from iSparrow import SparrowWatcher
+from .. import set_up_sparrow_env
+from copy import deepcopy
+
+
+class WatchFixture:
```

Comment:
This class concentrates all helper function that are needed in the tests and all the data that they need. Helper funcitons mostly live here because sonarcloud complains about code repetition if I just spell them out in the tests themselves.